### PR TITLE
Introduce term_exists_args filter

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket:  https://core.trac.wordpress.org/ticket/55439

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/taxonomy.php [hunk 1]
```diff
@@ -1559,6 +1559,18 @@ function term_exists( $term, $taxonomy = '', $parent = null ) {
 		$defaults['fields']   = 'all';
 	}
 
+	/**
+	 * Filter default arguments for checking if term exists.
```
Comment (audrasjb) [inline src/wp-includes/taxonomy.php]: Use third-person verbs for better consistency with other docblocks (= "Filters …")

**Review by audrasjb — CHANGES_REQUESTED** [review]

File: src/wp-includes/taxonomy.php (hunk 1 above)
Comment (spacedmonkey) [inline src/wp-includes/taxonomy.php]:
```suggestion
	 * Filters default arguments for checking if term exists.
```

Comment (spacedmonkey) [reply src/wp-includes/taxonomy.php]: Better?

File: src/wp-includes/taxonomy.php [hunk 2]
```diff
@@ -1559,6 +1559,18 @@ function term_exists( $term, $taxonomy = '', $parent = null ) {
...
 
+	/**
+	 * Filters default arguments for checking if term exists.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array      $defaults An array of arguments for term_exists.
+	 * @param int|string $term     The term to check. Accepts term ID, slug, or name.
+	 * @param string     $taxonomy Optional. The taxonomy name to use.
+	 * @param int        $parent   Optional. ID of parent term under which to confine the exists search.
```
Comment (peterwilsoncc) [inline src/wp-includes/taxonomy.php]:
```suggestion
	 * @param string     $taxonomy The taxonomy name to use. An empty string indicates the search is against all taxonomies.
	 * @param int|null   $parent   ID of parent term under which to confine the exists search. Null indicates the search is unconfined.
```

The variables aren't really optional, WP will always pass them if the developer requests them. Certain values have special meaning.

File: src/wp-includes/taxonomy.php [hunk 3]
```diff
@@ -1559,6 +1559,18 @@ function term_exists( $term, $taxonomy = '', $parent = null ) {
...
+	 */
+	$defaults = apply_filters( 'wp_term_exists_default_args', $defaults, $term, $taxonomy, $parent );
```
Comment (peterwilsoncc) [inline src/wp-includes/taxonomy.php]:
```suggestion
	$defaults = apply_filters( 'term_exists_default_args', $defaults, $term, $taxonomy, $parent );
```

It's more consistent with other parts of WP to match the function name, even if it doesn't have the WP prefix.

**Review by peterwilsoncc — COMMENTED** [review]
Why did you apply the filter to the default arguments rather that the calculated arguments?

The question behind the question, did you do this for your convenience or the convenience of developers extending WordPress?

**Comment by spacedmonkey** [comment]
> Why did you apply the filter to the default arguments rather that the calculated arguments?

Those default args and then used to generate args for 3 different get_terms calls. 1. for id, 2. for slug 3. for name. If I were to apply to all args, it would mean repeating the filter 3 times in the same function. Which feels weird and wrong. The value of the filter would be different as well. 

The ask here, was to add a filter so that developers can easily identify term_exists queries. Can do this, by changing the default args, maybe change the cache_domain or something. If you want to change any other args, you can do that with a filter like this `get_terms_arg`.

**Comment by peterwilsoncc** [comment]
> Those default args and then used to generate args for 3 different get_terms calls. 1. for id, 2. for slug 3. for name. If I were to apply to all args, it would mean repeating the filter 3 times in the same function. Which feels weird and wrong. The value of the filter would be different as well.

OK, that makes sense. Thanks.

Is it okay if we go with one last name change for the filter `term_exists_default_query_args` -- query args is a quite common suffix and I forgot to add it while editing yesterday.

File: src/wp-includes/taxonomy.php [hunk 4]
```diff
@@ -1559,6 +1559,18 @@ function term_exists( $term, $taxonomy = '', $parent = null ) {
...
+	 * Filters default arguments for checking if term exists.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array      $defaults An array of arguments for term_exists.
+	 * @param int|string $term     The term to check. Accepts term ID, slug, or name.
+	 * @param string     $taxonomy The taxonomy name to use. An empty string indicates the search is against all taxonomies.
+	 * @param int|null   $parent   ID of parent term under which to confine the exists search. Null indicates the search is unconfined.
+	 */
+	$defaults = apply_filters( 'term_exists_default_args', $defaults, $term, $taxonomy, $parent );
```
Comment (spacedmonkey) [inline src/wp-includes/taxonomy.php]:
```suggestion
	$defaults = apply_filters( 'term_exists_default_query_args', $defaults, $term, $taxonomy, $parent );
```

**Pull request closed without merging**